pandas/tests/extension/test_arrow.py

```diff
@@ -1491,6 +1516,325 @@ def test_where_series(self, data, na_value, as_frame, request, using_array_manag
+            all_arithmetic_operators
+            in (
+                "__sub__",
+                "__rsub__",
+            )
+            and pa.types.is_unsigned_integer(pa_dtype)
+            and not pa_version_under2p0
+        ):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    raises=pa.ArrowInvalid,
+                    reason=(
+                        f"Implemented pyarrow.compute.subtract_checked "
+                        f"which raises on overflow for {pa_dtype}"
+                    ),
+                )
+            )
+        elif arrow_temporal_supported:
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    raises=TypeError,
+                    reason=(
+                        f"{all_arithmetic_operators} not supported between"
+                        f"pd.NA and {pa_dtype} Python scalar"
+                    ),
+                )
+            )
+        elif (
+            all_arithmetic_operators in {"__rtruediv__", "__rfloordiv__"}
+            and (pa.types.is_floating(pa_dtype) or pa.types.is_integer(pa_dtype))
+            and not pa_version_under2p0
+        ):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    raises=pa.ArrowInvalid,
+                    reason="divide by 0",
+                )
+            )
+        op_name = all_arithmetic_operators
+        ser = pd.Series(data)
+        # pd.Series([ser.iloc[0]] * len(ser)) may not return ArrowExtensionArray
+        # since ser.iloc[0] is a python scalar
+        other = pd.Series(pd.array([ser.iloc[0]] * len(ser), dtype=data.dtype))
+        if pa.types.is_floating(pa_dtype) or (
+            pa.types.is_integer(pa_dtype) and all_arithmetic_operators != "__truediv__"
+        ):
+            monkeypatch.setattr(TestBaseArithmeticOps, "_combine", self._patch_combine)
+        self.check_opname(ser, op_name, other, exc=self.series_array_exc)
+
+    def test_add_series_with_extension_array(self, data, request):
+        pa_dtype = data.dtype.pyarrow_dtype
+        if (
+            not (
+                pa.types.is_integer(pa_dtype)
+                or pa.types.is_floating(pa_dtype)
+                or (not pa_version_under8p0 and pa.types.is_duration(pa_dtype))
+            )
+            or pa_version_under2p0
+        ):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    raises=NotImplementedError,
+                    reason=f"add_checked not implemented for {pa_dtype}",
+                )
+            )
+        super().test_add_series_with_extension_array(data)
+
+
+class TestBaseComparisonOps(base.BaseComparisonOpsTests):
+    def assert_series_equal(self, left, right, *args, **kwargs):
+        # Series.combine for "expected" retains bool[pyarrow] dtype
+        # While "result" return "boolean" dtype
+        right = pd.Series(right._values.to_numpy(), dtype="boolean")
```

Comment:
this is not great

Reply:
Yeah using `Series.combine` to generate `expected` for these ExtensionArray tests doesn't always compare nicely with pyarrow. 

I wouldn't say there's a bug in `Series.combine` or this implementation, just a consequence of `Series.combine` using Python comparisons semantics and not necessarily Arrow comparison semantics

Reply:
In general I would maybe also not do too much effort in trying to fully implement all those tests (too late probably) following the base extension tests exactly. As those need to be written in a generic way, that also makes them not very robust (eg the `combine()` method) / often awkward to write. 
It might make sense to write a set of custom, more targeted set of tests for a group of dtypes like the arrow dtypes (for certain operations)